Core/Services/OfferService.cs

```diff
@@ -44,14 +58,33 @@ public async Task<OfferCreateDTO> CreateOfferAsync(OfferCreateDTO offerCreate, s
             offer.OfferCreatorId = userId;
             offer.CreationDate = DateTimeOffset.UtcNow;
             offer.IsClosed = false;
-            offer.CreatorRoleId = _roleRepository.GetRoleByName(offerCreate.Role);
-            offer.GoodCategoryId = _goodCategoryRepository
+            offer.CreatorRoleId = _roleService.GetRoleByName(offerCreate.Role);
+            offer.GoodCategoryId = _goodCategoryService
                 .GetGoodCategoryByName(offerCreate.GoodCategory);
             offer.OfferPointId = await _pointService.CreatePointForOfferAsync(offerCreate.Point);
 
             await _offerRepository.InsertAsync(offer);
             await _offerRepository.SaveChangesAsync();
             return _mapper.Map<OfferCreateDTO>(offerCreate);
         }
+
+        public async Task<OfferInfoDTO> GetOfferByIdAsync(int offerId, string userId)
+        {
+            var offer = _offerRepository.Query().FirstOrDefault(o => o.Id == offerId && o.OfferCreatorId == userId);
+
+            ExceptionMethods.OfferNullCheck(offer);
+
+            var offerInfo = _mapper.Map<OfferInfoDTO>(offer);
+            offerInfo.Role = (await _roleRepository.GetByIdAsync(offer.CreatorRoleId)).Name;
+            offerInfo.GoodCategory = (await _goodCategoryRepository.
```

Comment:
I think we should check all entities from db for null

Reply:
done!